tokio/src/io/util/fill_buf.rs

```diff
@@ -40,6 +40,12 @@ impl<'a, R: AsyncBufRead + ?Sized + Unpin> Future for FillBuf<'a, R> {
                 // Safety: This is necessary only due to a limitation in the
                 // borrow checker. Once Rust starts using the polonius borrow
```

Comment:
while we're improving this comment it *might* also be nice to explain what the limitation in the borrow checker *is*, rather than just saying that there is one? not a blocker.